src/mappings/mappingUniquesHandlers.ts

```diff
@@ -1,28 +1,145 @@
+  item.owner = to.toString();
+  uniqueTransfer.itemId = item.id;
+  uniqueTransfer.collectionId = collection.id;
+
+  await collection.save();
+  await item.save()
   return uniqueTransfer.save();
 }
+
+export const handleUniquesMetadataSetEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug("uniqueMetadataSetEvent added: " + JSON.stringify(event.toHuman()));
+  const collectionId = event.event.data[0];
+  const itemId = event.event.data[1];
+  const data = event.event.data[2];
+  // const isFrozen = event.event.data[3];
+
+  const collection = await ensureCollection(collectionId.toString());
+  const item = await ensureItem(collectionId.toString(), itemId.toString());
+  item.metadataCid = data.toHuman().toString();
+  item.collectionId = collection.id;
+  await collection.save()
+  return item.save();
+}
+
+export const handleUniquesCollectionMetadataSetEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "uniqueCollectionMetadataSetEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const data = event.event.data[1];
+  // const isFrozen = event.event.data[2];
+
+  const collection = await ensureCollection(collectionId.toString());
+  collection.metadataCid = data.toHuman().toString();
+
+  return collection.save();
+}
+
+export const handleUniquesDestroyedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesDestroyedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+
+  const collection = await ensureCollection(collectionId.toString());
+  collection.isDestroyed = true;
+  return collection.save()
+}
+
+export const handleUniquesBurnedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesBurnedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const itemId = event.event.data[1];
+
+  const item = await Item.get(itemId.toString());
+  item.isBurned = true;
+  return item.save();
+}
+
+export const handleUniquesIssuedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesIssuedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const itemId = event.event.data[1];
+  const owner = event.event.data[2];
+
+  const collection = await ensureCollection(collectionId.toString());
+  const item = await ensureItem(collectionId.toString(), itemId.toString());
+  item.owner = owner.toString();
+  item.collectionId = collection.id;
+  item.isBurned = false;
+  await collection.save()
+  return item.save();
+}
+
+export const handleUniquesCreatedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesCreatedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+
+  const collection = await ensureCollection(collectionId.toString());
+  collection.isDestroyed = false;
+  return collection.save()
```

Comment:
I would suggest we better highlight the unexpected cases. Here is one way of doing it for this case:
```suggestion
  let collection = await Collection.get(collectionId);
  if (collection) {
    logger.error('Handled UniquesCreatedEvent for a collection which already exists ', collectionId);
  }
  else {
    collection = await ensureCollection(collectionId);
  }
  return collection.save();
```

Reply:
Yes, this case is managed by the fn `ensureCollection`

Reply:
What I'm suggesting here is logging the error based on our expectation. Inside `ensureCollection` you wouldn't know if the expectation is that the collection should have existed or quite the opposite.